docs/api/ethereum.md

```diff
@@ -1030,6 +1045,256 @@ A class instance `variable` holding a `string` which represents smart contract a
+6. Taker performs the deploy.
+
+`Deploy` class is responsible for defining what will happen in the atomic swap. We support two different deploy configurations which we will call a fixed deploy and a dynamic deploy.
+
+In fixed deploy, the Taker of the deploy (its wallet address) is known, and we want to make an atomic deploy specifically with him and only him. For this, we need to set `deploy.takerId`.
+
+In dynamic deploy, we do not care who performs the deploy. In this case, we do not set `deploy.takerId`. Now any account (wallet) will be able to perform such deploy and will automatically become its Taker.
+
+### DeployGateway(provider, deployGatewayId)
+
+A `class` which represents a smart contract on the Ethereum blockchain.
+
+**Arguments**
+
+| Argument | Description
+|-|-
+| deployGatewayId | [required] A `string` representing an address of the [0xcert deploy gateway smart contract](#public-addresses) on the Ethereum blockchain.
+| provider | [required] An instance of an provider.
+
+**Usage**
+
+```ts
+import { MetamaskProvider } from '@0xcert/ethereum-metamask-provider';
+import { DeployGateway } from '@0xcert/ethereum-deploy-gateway';
+
+// arbitrary data
+const provider = new MetamaskProvider();
+const deployGatewayId = '0x073d230a53bffc8295d9a5247296213298e3fbcf';
+
+// create deploy gateway instance
+const deployGateway = new DeployGateway(provider, deployGatewayId);
+```
+
+### cancel(deploy)
+
+An `asynchronous` class instance `function` which marks the provided `deploy` as canceled. This prevents the `deploy` to be performed.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| deploy.assetLedgerData.capabilities | [required] An array of `AssetLedgerCapability` representing the capabilities of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.name | [required] A `string` representing the name of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.owner | [required] A `string` representing the address of owner of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.schemaId | [required] A `string` representing the schemaId of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.symbol | [required] A `string` representing the symbol of the `AssetLedger` that will be deployed.
+| deploy.assetLedgerData.uriBase | [required] A `string` representing the uriBase of the `AssetLedger` that will be deployed.
+| deploy.expiration | [required] An `integer` number representing the timestamp in milliseconds at which the deploy expires and can not be performed any more.
+| deploy.makerId | [required] A `string` representing the Ethereum account address which makes the deploy. It defaults to the `accountId` of a provider.
+| deploy.seed | [required] An `integer` number representing the unique deploy number.
+| deploy.takerId | A `string` representing the Ethereum account address which will be able to perform the deploy on the blockchain. This account also pays for the gas cost.
```

Comment:
| deploy.takerId | A `string` representing the Ethereum account address that will be able to perform the deploy on the blockchain. This account also pays for the gas cost.